proposals/lip-0047.md

```diff
@@ -498,11 +498,147 @@ This module does not define any specific logic for other modules.
 
 This module does not define any endpoint for off-chain services.
 
-### Protocol Logic During Block Lifecycle
+### Genesis Block Processing
 
-#### After Genesis Block Execution
+The following two steps are executed as part of the genesis block processing, see the “Update genesis block schema and processing” LIP for details. 
 
-After the genesis block `g` is executed, the following logic is executed:
+#### Genesis State Initialization
+
+To initialize the state of a genesis block `g`, the following logic is executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the bytes included in the block assets for the PoA module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisPoAStoreSchema` schema, given below, then: 
+
+* Checks for the `genesisBlockAssetObject.validatorSubstore`: 
```

Comment:
Wouldn't it be easier here and also better encourage code re-use to write:
For each entry in `genesisBlockAssetObject.validatorSubstore` perform the verification and execution logic of the authority registration command with the same parameters as input and `address` being the address of the sender account. Only the last step in the execution, callling `registerValidatorKeys`, is not done here as API calls should only be done in the genesis state finalization.

Reply:
I added a clarification sentence in this direction. This way we have the point of re-use, the strictly correct specs for the genesis and I don't need to revert changes  that are actually correct.